docs/usage.md

```diff
@@ -0,0 +1,65 @@
+# Usage
+
+## Rendering
+
+### Basic HTML rendering
+
+The demo project comes with 2 files `src/html.htl` and `src/html.pre.js` which renders HTML from the requested Markdown file. The `htl` file is the rendering template while the `pre.js` allows to interact with the payload before it gets rendered by the template.
+The These files are named based on the following logic: `<selector>_<extension>.htl` and `<selector>_<extension>.pre.js`. In this case, the `selector` is undefined. They will handle all requests coming to `http://localhost:3000/**/*.html`.
+
+The `htl` file format must respect the HTL spec, check the [htlengine](https://github.com/adobe/htlengine) for more details.
+The `pre.js` file must only export a `pre` function, check the [helix-pipeline](https://github.com/adobe/helix-pipeline) for the method signature.
+
+### Custom selector
+
+You can define the following pair `src/myselector_html.htl` and `src/myselector_pre.js` to handle request like `http://localhost:3000/**/*.myselector.html`. You can create as many selector as you want.
+
+One of the usage of the selectors (in combination with ESI - [Edge Server Includes](https://www.w3.org/TR/esi-lang)) could be to have multiple HTML rendering for your site. You could have 3 `htl` files (together with their `pre.js`):
+
+1. `html.htl`
+1. `homepage_html.htl`
+1. `others_html.htl`
+
+The `html.htl` file could look like this:
+
+```html
+<esi:include src="${payload.dispatch_url}" />
```

Comment:
do we really want to promote this technique ?

Reply:
We do not have a better way for now to have 2 different renderings for one site.

Reply:
true. somehow I don't like the extra roundtrip for the ESI include. ideally, the template selection logic, especially in this case when it is so simple, should happen in fastly....

Reply:
yes, maybe but then it cannot be the selector. In any case, this is how it is today, thus documented as is. When we have something better, we update the doc!